libs/output-mapping/tests/Storage/TableStructureValidatorTest.php

```diff
@@ -0,0 +1,464 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\OutputMapping\Tests\Storage;
+
+use Keboola\OutputMapping\Exception\InvalidOutputException;
+use Keboola\OutputMapping\Exception\InvalidTableStructureException;
+use Keboola\OutputMapping\Mapping\MappingFromConfigurationSchemaColumn;
+use Keboola\OutputMapping\Storage\TableStructureValidator;
+use Keboola\OutputMapping\Tests\AbstractTestCase;
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\ClientException;
+
+class TableStructureValidatorTest extends AbstractTestCase
+{
+    public function testValidateTableStructure(): void
+    {
+        $schema = [
+            new MappingFromConfigurationSchemaColumn([
+                'name' => 'col1',
+                'data_type' => [
+                    'base' => [
+                        'type' => 'STRING',
+                        'length' => '255',
+                    ],
+                    'snowflake' => [
+                        'type' => 'VARCHAR',
+                        'length' => '255',
+                    ],
+                ],
+            ]),
+            new MappingFromConfigurationSchemaColumn([
+                'name' => 'col2',
+                'data_type' => [
+                    'base' => [
+                        'type' => 'NUMERIC',
+                    ],
+                ],
+            ]),
+            new MappingFromConfigurationSchemaColumn([
+                'name' => 'col3',
+                'data_type' => [
+                    'base' => [
+                        'type' => 'NUMERIC',
+                        'length' => '123',
+                    ],
+                    'snowflake' => [
+                        'type' => 'INT',
+                        'length' => '123',
+                    ],
+                ],
+            ]),
+        ];
+
+        $validator = new TableStructureValidator(true, $this->getClientMock());
+        $validator->validateTable('in.c-main.table', $schema);
+
+        self::assertTrue(true);
+    }
+
+    public function testSkipValidationWithoutFeature(): void
+    {
+        $schema = [];
+        $validator = new TableStructureValidator(false, $this->getClientMock());
+        $validator->validateTable('in.c-main.table', $schema);
+
+        self::assertTrue(true);
+    }
+
+    public function testSkipValidationIfTableNotExists(): void
+    {
+        $clientMock = $this->createMock(Client::class);
+        $clientMock
```

Comment:
```suggestion
        $clientMock
            ->expects(self::once())
```

Mockům bys měl nastavovat pro metody i to, kolikrát je očekávaný že budou zavolané.

Pokud bys do `validateTable` hodil hned na začátek return, tak ti tvůj test bez toho také projde.

Reply:
https://github.com/keboola/platform-libraries/pull/266/commits/f2ba8a1343a64860fdab1a418dd3ada29081d935